python/main.py

```diff
@@ -18,18 +21,54 @@
     allow_headers=["*"],
 )
 
+#import json
+with open("items.json") as f:
+    items = json.load(f)
+items_list = items["items"]
+
 @app.get("/")
 def root():
     return {"message": "Hello, world!"}
 
 @app.post("/items")
-def add_item(name: str = Form(...)):
+async def add_item(name: str = Form(...),
+                category: str = Form(...),
+                image: UploadFile = File(...)):
+    image_bytes = await image.read() # turn the image into a buffer-like object
+    image_hashed = hashlib.sha256(image_bytes).hexdigest()  # hash the image
```

Comment:
How do we know if this is a valid image at this point? What happens if you try to upload a `.txt` or `.json` file?

Reply:
fixed! I added following codes.

    ALLOWED_IMAGE_EXTENSIONS = ['.jpg', '.jpeg', '.png']
    def is_image_file(filename: str) -> bool:
        extension = os.path.splitext(filename)[1].lower()
        return extension in ALLOWED_IMAGE_EXTENSIONS
    if not is_image_file(image.filename):
        return {"error": "Invalid image file"}

Reply:
Nice! Similar to the another comment I made, this message `{"error": "Invalid image file"}` will be given a 200 OK status code.

Is there a status code that might be more suitable for this situation?

Reply:
fixed!